invokeai/app/api/routers/model_records.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2023 Lincoln D. Stein
+"""FastAPI route for model configuration records."""
+
+
+from hashlib import sha1
+from random import randbytes
+from typing import List, Optional
+
+from fastapi import Body, Path, Query, Response
+from fastapi.routing import APIRouter
+from pydantic import BaseModel, ConfigDict, TypeAdapter
+from starlette.exceptions import HTTPException
+
+from invokeai.app.services.model_records import DuplicateModelException, InvalidModelException, UnknownModelException
+from invokeai.backend.model_manager.config import AnyModelConfig, BaseModelType, ModelType
+
+from ..dependencies import ApiDependencies
+
+model_records_router = APIRouter(prefix="/v1/model_records", tags=["model_records"])
+
+ModelConfigValidator = TypeAdapter(AnyModelConfig)
+
+
+class ModelsList(BaseModel):
+    """Return list of configs."""
+
+    models: list[AnyModelConfig]
+
+    model_config = ConfigDict(use_enum_values=True)
+
+
+ModelsListValidator = TypeAdapter(ModelsList)
+
+
+@model_records_router.get(
```

Comment:
Why is this a separate router from the models router? If one is for DB and one is for yaml, it may be simpler to just have one router with a flag while this is in development. Once this is fully rolled out, the front-end will just be calling one or the other, right?

Reply:
The new routes are where the new API is going, not to distinguish db from sql. My plan is to build a side-by-side set of routes that are easily distinguished from the original ones, have the frontend switch over to using the new ones, them remove the old calls will be removed.

In response to your comment I've moved the `model_records` calls back into `models` and have changed the paths to look like this:

#### List model configurations
```
GET /api/v1/model/record
```

#### Fetch a model by its configuration ID:
```
GET /api/v1/model/record/i/{id}
```

### Install models:
```
POST /api/v1/model/install/
```
### Convert model
```
PATCH /api/v1/model/convert/{id}
```